src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs

```diff
@@ -115,4 +115,14 @@ public interface IFileStore
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns></returns>
     Task CopyFileAsync(long originalVersion, long newVersion, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously sets metadata on a blob.
+    /// </summary>
+    /// <param name="version">The DICOM instance version.</param>
+    /// <param name="metadata">Metadata to store.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A string representing eTag of changed blob.</returns>
+    /// <remarks>Used only for integration tests today to ensure we capture eTag changing logic even when contents of file not changed.</remarks>
+    Task<string> SetInstanceBlobMetadataAsync(long version, IDictionary<string, string> metadata, CancellationToken cancellationToken = default);
```

Comment:
Where is this used? I thought in design we decided not to use BlobMetadata because it is hard to access.

Reply:
I'm a bit apprehensive about logic only used for testing, like a method. Can we not mock this in another way?